### PR TITLE
Pin pyarrow<13 on macOS to avoid crash (sc-51866)

### PR DESCRIPTION
**Pull request by ihnorton**
Temporarily pin pyarrow<13 on macOS to avoid crash (sc-51866).

### CODE REVIEW

**Review by gspowley — APPROVED** [review]

**Comment by jdblischak** [comment]
> This does not appear to be possible for 2.25

I confirmed this. The latest tiledb that can be co-installed with pyarrow 12 is 2.24.1. The reason is the aws-crt-cpp pin (though there may be additional incompatible pins; this is the just the first incompatibility the solver finds). pyarrow 12 was last built against aws-crt-cpp 0.26, but tiledb 2.25 was first built against aws-crt-cpp 0.27.

```sh
mamba create --dry-run -n test -c conda-forge 'pyarrow<13' tiledb
## + aws-crt-cpp                    0.26.12  he940a02_1            conda-forge      340kB
## + tiledb                          2.24.1  h2aae180_1            conda-forge        4MB
## + python                          3.11.9  hb806964_0_cpython    conda-forge       31MB
## + libarrow                        12.0.1  h0de2578_50_cpu       conda-forge       28MB
## + pyarrow                         12.0.1  py311h8eb17fa_50_cpu  conda-forge        4MB

# pyarrow 12 was not built against aws-crt-cpp 0.27
mamba create --dry-run -n test -c conda-forge 'pyarrow<13' aws-crt-cpp=0.27
# long solver error

# tiledb 2.25 was not built against aws-crt-cpp 0.26
mamba create --dry-run -n test -c conda-forge tiledb=2.25 aws-crt-cpp=0.26
## The following packages are incompatible
## ├─ aws-crt-cpp 0.26**  is requested and can be installed;
## └─ tiledb 2.25**  is not installable because it requires
##    └─ aws-crt-cpp >=0.27.3,<0.27.4.0a0 , which conflicts with any installable versions previously reported.
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -79,7 +80,8 @@ outputs:
         - numpy
         - libtiledbvcf {{ version }}
         - python
-        - pyarrow
+        - pyarrow <13                            # [osx and py<312]
+        - pyarrow                                # [not osx and py>=312]
```
Comment (jdblischak) [inline recipe/meta.yaml:84]:
```suggestion
        - pyarrow                                # [not osx or py>=312]
```

https://en.wikipedia.org/wiki/De_Morgan%27s_laws

**Comment by jdblischak** [comment]
Wait a second. I realized I don't understand the context of this PR. What is the crash on macOS? Has a new error been identified?

I know in the past that we pinned pyarrow in order to avoid a macOS-specific segfault due to the AWS SDK (https://github.com/apache/arrow/issues/42154), but it's my understanding that this bug only affects the PyPI wheels because both tiledb and pyarrow bundle statically linked AWS C libraries.

xref: https://github.com/TileDB-Inc/tiledb-vcf-feedstock/pull/114, https://github.com/TileDB-Inc/tiledb-vcf-feedstock/pull/111

**Comment by ihnorton** [comment]
Closing, @gspowley is working on the underlying issue.

**Pull request closed without merging**